### ISSUE TITLE
AttributeError: settings

### ISSUE THREAD

**Issue by donthuanalyst** [blobtoolkit/blobtools2#34]
Hello,

I am getting the following error while running blobtools2. Could you please help me to fix this error.

Thank you for your help.

blobtools create \
        --fasta /work/assemblies/asm.v2.FINAL.fasta \
        --taxid 7460 \
        --taxdump /work/databases/taxdump \
        --hits /work/projects/blobtools/results/diamond_asm_vs_uniprotRefProteomes_07_07_2021/diamondF15_asm_vs_upRefprot
eomes.out \
        --hits /work/projects/blobtools/results/blastn_asm_vs_nt_07_06_2021/blastn_asm_vs_nt.out \
        --taxrule bestsumorder \
        --cov /work/projects/blobtools/results/bwa_allIlmn_vs_Asm_06_28_2021/ilmn_vs_asm.paired.sorted.bam \
        --threads 32 \
        HBasm

**Error message**
<pre>
Loading sequences from /work/tgiray/rdonthu/HB_assemblies/asm.v2.FINAL.fasta
Loading mapping data from /work/tgiray/rdonthu/projects/PR_blobtools/results/bwa_PRallIlmn_vs_PRAsm_06_28_2021/ilmn_vs_asm.paired.sorted.bam as
 PRilmn_vs_PRasm.paired.sorted
Parsing taxdump
bestsumorder
/workprojects/blobtools/results/diamond_asm_vs_uniprotRefProteomes_07_07_2021/diamondF15_asm_vs_upRefproteomes.out
Traceback (most recent call last):
  File "/cm/shared/apps/kiran/envs/kiran/blob/blobtools2/bin/lib/add.py", line 194, in <module>
    main()
  File "/cm/shared/apps/kiran/envs/kiran/blob/blobtools2/bin/lib/add.py", line 147, in main
    parsed = field["module"].parse(
  File "/cm/shared/apps/kiran/envs/kiran/blob/blobtools2/lib/hits.py", line 526, in parse
    bins = bin_hits(
  File "/cm/shared/apps/kiran/envs/kiran/blob/blobtools2/lib/hits.py", line 154, in bin_hits
    windows = set_windows(meta, taxrule)
  File "/cm/shared/apps/kiran/envs/kiran/blob/blobtools2/lib/hits.py", line 117, in set_windows
    if "stats_windows" in meta.settings:
AttributeError: settings
</pre>

**Comment by pstrope** [comment]
I am also getting the same error:
```
Loading sequences from ./BTK_TUTORIAL/FILES/ASSEMBLY_NAME.fasta
 - processing LM525554.1: : 587it [00:03, 189.34it/s]
Loading parsed taxdump
Traceback (most recent call last):
  File "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/home/stropepk/BTK/blobtoolkit/blobtools2/lib/add.py", line 194, in <module>
    main()
  File "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/home/stropepk/BTK/blobtoolkit/blobtools2/lib/add.py", line 147, in main
    parsed = field["module"].parse(
  File "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/home/stropepk/BTK/blobtoolkit/blobtools2/lib/hits.py", line 519, in parse
    bins = bin_hits(
  File "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/home/stropepk/BTK/blobtoolkit/blobtools2/lib/hits.py", line 154, in bin_hits
    windows = set_windows(meta, taxrule)
  File "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/home/stropepk/BTK/blobtoolkit/blobtools2/lib/hits.py", line 117, in set_windows
    if "stats_windows" in meta.settings:
AttributeError: settings
```

**Comment by rjchallis** [comment]
Thanks for pointing this out - I'd added a configuration option and hadn't realised that it would break things when not set. I've just pushed a commit to the develop branch that should fix this and will include it in the next release at the end of the week.

**Comment by gchchung** [comment]
Any way to work around the hits.py line 117 AttributeError at this time?

**Comment by rjchallis** [comment]
This issue should be fixed in the latest release (v2.6.2). An alternate workaround with v2.6.1 would be to add a metadata file with a settings section using `blobtools create/add`
